pkg/util/goroutinemap/goroutinemap.go

```diff
@@ -73,4 +80,9 @@ func (grm *goRoutineMap) operationComplete(operationName string) {
 	grm.Lock()
 	defer grm.Unlock()
 	delete(grm.operations, operationName)
+	grm.wg.Done()
```

Comment:
`Done()` should be called with `defer`. Defer statements are executed in stack order (last in, first executed), so make sure `defer grm.wg.Done()` is called after `defer grm.Unlock()`.